src/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -162,6 +162,25 @@ public int findUnitsTransportCost(final List<Unit> units) {
     return transportCost;
   }
 
+  public List<Unit> getAirThatCantLandOnCarrier(final PlayerID player, final Territory t, final List<Unit> units) {
+    final GameData data = ai.getGameData();
+    int capacity = AirMovementValidator.carrierCapacity(units, t);
+    final Collection<Unit> airUnits = Match.getMatches(units, ProMatches.unitIsAlliedAir(player, data));
+    List<Unit> airThatCantLand = new ArrayList<Unit>();
+    for (final Unit airUnit : airUnits) {
+      final UnitAttachment ua = UnitAttachment.get(airUnit.getType());
+      final int cost = ua.getCarrierCost();
```

Comment:
- It wasn't obvious on the first or even second pass of reading this code. I think you benefit   from too much context/familiarity with this code, and may have a different perspective a years time from now.  
- Assuming each "cost" variable refers to the same thing is not a wise assumption. So this is actually burden on the reader to check that since it is not obvious which type of cost is referred to without more context.  Requiring more context, looking at other methods - all to understand this public method is a smell. You should be able to understand a public method quite readily by reading just it. Questions as to what things represent, and needing to look up more information is an indication that the code is not self explanatory.